mono/metadata/threadpool-worker-default.c

```diff
@@ -8,6 +8,12 @@
  * Licensed under the MIT license. See LICENSE file in the project root for full license information.
  */
 
+#ifdef __cplusplus
```

Comment:
Unnecessary now -- there is a list of invalid C++ files in Makefile.am, just a few, but too many to give rationale for.